### PR TITLE
Manual shutdown for worklock

### LINKED ISSUE #1776: Provide for manual shutdown of WorkLock contract by allowing shutdown() to be owner-callable.

### PR DESCRIPTION
**Pull request by vzotova**
Fixes #1776
Opens #1787

### CODE REVIEW

File: nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol [hunk 1]
```diff
@@ -338,11 +338,22 @@ contract WorkLock {
...
+    /**
+    * @notice Cancels distribution, makes possible to retrieve all bids and owner gets all tokens
+    */
+    function internalShutdown() internal {
+        startBidDate = 0;
+        endBidDate = 0;
         endCancellationDate = uint256(0) - 1; // "infinite" cancellation window
-        token.safeTransfer(creator, tokenSupply);
+        token.safeTransfer(owner(), tokenSupply);
+        emit Shutdown(msg.sender);
```
Comment (derekpierre) [inline nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol]: 🎉

**Comment by michwill** [comment]
Am I right that all bidders would get their ETH back as a result of shutdown / cancelling bids? (should it be in tests?)

**Comment by vzotova** [comment]
> Am I right that all bidders would get their ETH back as a result of shutdown / cancelling bids? (should it be in tests?)

Shutdown makes cancellation phase infinite, so each bidder can call `cancelBid` anytime (and this is in tests)

**Comment by michwill** [comment]
Ahh, right, and cancelBid is tested separately :+1:

File: nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol [hunk 2]
```diff
@@ -338,11 +338,22 @@ contract WorkLock {
     }
 
     /**
-    * @notice Cancels distribution, makes possible to retrieve all bids and creator gets all tokens
+    * @notice Cancels distribution, makes possible to retrieve all bids and owner gets all tokens
     */
-    function shutdown() internal {
+    function shutdown() external onlyOwner {
+        require(!isClaimingAvailable(), "Claiming has already been enabled");
```
Comment (KPrasch) [inline nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol]: s the any mitigation or follow up for this condition?

Comment (vzotova) [reply nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol]: Why need any mitigations?

**Pull request merged**